assets/scripts/ui/Tooltip.jsx

```diff
@@ -53,6 +56,10 @@ function Tooltip ({
       </Tippy>
     )
   }
+
+  // If something happens and above conditions can't render, just
```

Comment:
does this have/need test coverage then? I'm not sure what expected behavior would be in this case.

I'm not sure I really understand the separate if statements for source & target either.

Reply:
The separate statements for source and target has to do with the way Tippy implements its singleton behavior. It's unfortunate; they are fundamentally different concepts (IMO) but they're overloading a single component to handle both situations. (It may be easier for them to do it this way; not sure.)

I had considered creating our own abstraction over Tippy, but ultimately decided against it. I don't think we need to put another mental model on top of Tippy, which makes trying to use Tippy itself harder.

Reply:
To answer the first question: I think I should cover it with tests, yes. There is no use case where passing the children through is desirable, but in the event that it _does_ happen, this is an escape hatch (a React component must always return _something_).

Reply:
Following up on this (I edited the above comment to cross out a statement I'm deprecating). Although we don't use non-singleton tooltips right now, the "escape hatch" final return should just be a non-singleton tooltip. This is the "default" tooltip that Tippy offers and that's the tooltip that happens when no `source` or `target` prop is provided (e.g. the parent component doesn't use the `useSingleton()` hook).